tsig.go

```diff
@@ -39,6 +39,14 @@ type TSIG struct {
 	OtherData  string `dns:"size-hex:OtherLen"`
 }
 
+// TsigProvider provides the API to plug-in a custom TSIG implementation.
+type TsigProvider interface {
+	// Generate is passed the DNS message to be signed and the partial TSIG RR. It returns the signature and nil, otherwise an error.
+	Generate([]byte, *TSIG) ([]byte, error)
```

Comment:
These parameters should be named, so something like: `(msg []byte, rr *TSIG)` or whatever is appropriate. That's a little clearer for people. Same with `Verify` below.

Reply:
Done.